patches/post/20235630-device-pagination.patch

```diff
@@ -0,0 +1,58 @@
+diff --git a/metal/v1/api_devices.go b/metal/v1/api_devices.go
```

Comment:
If we were to maintain local copies of the Go templates, we could update the template files whenever we update the OpenAPITools version via the `meta` CLI command which dumps the templates. In that case, is this where we would patch in a generic (mustache) version of the paginated executor:

https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/go/api.mustache#L77-L79

Reply:
Finding a generic approach will be helpful for templating our Generated Go pattern across Fabric, NE, and future Go SDKs.

Reply:
For Metal, if this is safe for unpaginated endpoints, perhaps we can always include `ExecuteWithPagination` and it will degrade to a less efficient `Execute`:

`if page.Meta.GetLastPage() <= page.Meta.GetCurrentPage() {`

Reply:
```
type Paginated interface {
  GetCurrentPage() int32
  GetLastPage() int32
}

...
if meta, ok := page.Meta.(Paginated); meta != nil && ok { // check meta first? does meta always exist? A GetMeta() Interface could be used to check for it
  do the looping
}
```